File: design-patterns/go/state/order_state/order_state.go
// order_state.go

package main

type IOrderState interface {
	Process()
}

type OrderState struct {
	context *OrderContext
	IOrderState
}

func NewOrderState(iOrderState IOrderState, context *OrderContext) (orderState *OrderState) {
	orderState = &OrderState{}
	orderState.IOrderState = iOrderState
	orderState.context = context

	context.SetState(orderState)

	return
}

File: design-patterns/php/state/order/OrderState.php
<?php
// OrderState.php

namespace BigBoxCode\DesignPattern\State\Order;

abstract class OrderState {

    protected OrderContext $context;

    public function __construct(OrderContext $context) {
        $this->context = $context;

        $this->context->setState($this);
    }

    abstract function process(): void;
}
